src/main/java/com/github/splendor_mobile_game/game/model/Game.java

```diff
@@ -163,4 +187,29 @@ private Deck getRandomCards(CardTier tier, int amount) {
         return array;
     }
 
+    private ArrayList<Noble> getRandomNobles(int amount){
+        ArrayList<Noble> nobles = this.database.getAllNobles();
+
+        // We draw cards until deck will be empty
+        if ( nobles.size() < amount) amount=nobles.size();
+
+        ArrayList<Noble> array = new ArrayList<Noble>();
+
+        Random rand = new Random();
+        while(amount > 0) {
+            int index = rand.nextInt(nobles.size()); // Get random index
+            Noble drawnNoble=nobles.get(index);
+            if(array.contains(drawnNoble)){
+                continue;
+            }        
+            array.add(drawnNoble);
+                     
+            Log.DEBUG("Noble tile has been drawned of index: "+index);
+
+            amount--;
+        }
+
+        return array;
+    }
```

Comment:
You can now simplify the `getRandomNobles` method. We don't need to check for duplicates. For example:
```java
public static ArrayList < Integer > foo(ArrayList < Integer > tablica, int amount) {
    ArrayList < Integer > temp_tablica = new ArrayList < > ();
    Random rand = new Random();

    while (amount > 0 && tablica.size() > 0) {
        int index = rand.nextInt(tablica.size());
        temp_tablica.add(tablica.get(index));
        tablica.remove(index);
        amount--;
    }

    return temp_tablica;
}
```